api/tests/Feature/ExperienceTest.php

```diff
@@ -82,18 +79,26 @@ public function testSkillRelationshipsWorkWithPivot(): void
                     }
                 }
             }
-        ',
+        GRAPHQL,
             [
                 'id' => $this->platformAdmin->id,
             ]
-        );
-        // Assert that the experience  from query with all three skills, and that the pivot details work correctly.
-        $response->assertJson(fn (AssertableJson $json) => $json->has('data.user.workExperiences.0', fn (AssertableJson $json) => $json->where('id', $experience->id)
-            ->has('skills', 3, fn (AssertableJson $json) => $json->where('id', $userSkills[0]->skill_id)
-                ->where('experienceSkillRecord.details', 'first skill')
-            )
-        )
-        );
+        )->assertJsonFragment([
+            'id' => $userSkills[0]->skill->id,
+            'experienceSkillRecord' => [
+                'details' => 'first skill',
+            ],
+        ])->assertJsonFragment([
+            'id' => $userSkills[1]->skill->id,
+            'experienceSkillRecord' => [
+                'details' => 'second skill',
+            ],
+        ])->assertJsonFragment([
+            'id' => $userSkills[2]->skill->id,
+            'experienceSkillRecord' => [
+                'details' => 'third skill',
+            ],
+        ]);
```

Comment:
We've lost the assertion that there are three skills.  The test will now miss if, for example, four skills come back.

Reply:
An alternate approach might be ot use assertEqualCanonicalizing which will sort the arrays before comparison.
https://www.geeksforgeeks.org/phpunit-assertequalscanonicalizing-function/

Reply:
great suggestion! Updated in [update to use assert canonicalizing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11492/commits/317403eeda9b769731b421e4da102f406723e622)